dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -418,9 +459,22 @@ void ExchangeReceiverBase<RPCContext>::setUpConnection()
                 [this](bool meet_error, const String & local_err_msg) {
                     this->connectionDone(meet_error, local_err_msg, exc_log);
                 },
+                [this]() {
+                    this->connectionLocalDone();
+                },
+                [this]() {
+                    this->setLocalAlive();
+                },
                 ReceiverChannelWriter(&(getMsgChannels()), req_info, exc_log, getDataSizeInQueue(), ReceiverMode::Local));
 
-            rpc_context->establishMPPConnectionLocal(req, req.source_index, local_request_handler, enable_fine_grained_shuffle_flag);
+            rpc_context->establishMPPConnectionLocal(
+                req,
+                req.source_index,
+                local_request_handler,
+                enable_fine_grained_shuffle_flag,
+                [this]() {
+                    this->addLocalConnectionNum();
```

Comment:
Why need to `addLocalConnectionNum` after we have `setLocalAlive`?

Reply:
> Why need to `addLocalConnectionNum` after we have `setLocalAlive`?

I will combine them together